static/js/condense.js

```diff
@@ -40,9 +40,9 @@ exports.uncollapse = function (row) {
         content.removeClass("collapsed");
 
         if (message.condensed === true) {
-            // This message was condensed by the user, so re-show the
-            // [More] link.
-            condense_row(row);
+            // This message is not long, so
+            // hide the [More..]
+            row.find(".message_expander").hide();
```

Comment:
Can you explain your reasoning on why this is the correct change?

Reply:
@timabbott sure!
(I will refer to large messages as LM(>60% viewport)  and other messages as OM.)
The code was alright for LM and still works as such.

#why I removed what I removed
`condense_row(row);`
In either case, the `condensed` class was being added to the message container (which is needed only with LM since it follows collapse-condense-normal).
So, a `[More...]` link was being added to the messages every time.
Now, the aforementioned class is never added to OM and this basically solved the problem.

#why I added what I added
`row.find(".message_expander").hide();`
What it is doing is to not execute the whole function that was present but instead just a small part from it. Adding this line takes care of the removal of `[More...]` link, even if it is added by hotkey action or otherwise. This seems like the best approach to me without modifying other functionality in the code.
About the comment that was present before, it kind of made it all-inclusive and hence the iterator never reached beyond that particular `if` statement.

Rest assured I have tested this for multiple combinations of clicking and using the hotkey on messages of all sizes.

Reply:
@timabbott just a friendly reminder to please give some feedback on this.

Reply:
update: The comment felt confusing, so I simplified it.

Reply:
@timabbott It would be great if you could give me some feedback on if I need to improve on this piece of code, or if anything else is going wrong. Thanks again.